### DISCUSSION TITLE
ims.ims_user_agent_string

### DISCUSSION THREAD

**Discussion by AHMNco** [kyujin-cho/pixel-volte-patch#232] [Q&A]
Hi Dear please help me to use Samsung ims.ims_user_agent_string = #MANUFACTURER#_#MODEL#_Android#AV#_#BUILD# in my device, as you know Samsung device all around the world and using Samsung ims has big chance to use VoLTE on all devices, I have Motorola Edge 30 Ultra / Moto X30 Pro, with your Pixel IMS working only for Vodafone and other carriers not working, but if you help me to find Samsung ims user agent string I can apply instead of LG default you have in this app, or give me by yourself Samsung ims user agent for any popular models to apply in my device, I'm so frustrated and need your help
thanks a lot

**Comment by marsdoe** [comment]
+1
I want  to change Useragent
Now useragent is  Unknown and not availible for editing

**Reply by AHMNco** [reply]
Thank you for your answer, but my question is what is the user agent for Samsung Galaxy S22, because Samsung S22 is ready to use VoLTE and it's easy to use this phone User Agent instead of my Moto X30 Pro, I believe with this change my Phone VoLTE will be active, please help me how can I find User Agent string of Samsung Galaxy S22 or give it to me by your self, since you know LG User Agent String for sure your knowledge so much better than mine, thanks for your help in advance

**Comment by AHMNco** [comment]
I hope get an answer here, please someone expert in this field help me

**Reply by fillwithjoy1** [reply]
I don't have a Samsung, but a Google device should also do the job

User Agent: `Google_Pixel 6 Pro_Android 14_UQ1A.240205.002`
If that works for you please thumbs up or leave a comment on the #274 pull request

**Reply by AHMNco** [reply]
Thank you but it's not working, Google Pixel already less than Motorola phones, so maybe operator didn't add Google Pixel too

**Reply by fillwithjoy1** [reply]
What's your operator? If it's vodafone, they should have already implemented Google devices for ims.

**Reply by AHMNco** [reply]
no, it's not Vodafone, my Moto already working with Vodafone, it's Turkcell, and also MCI

**Reply by fillwithjoy1** [reply]
Okay, I don't have a samsung device so I can't help you there :(
Good luck though

**Reply by AHMNco** [reply]
Thank you

**Comment by AHMNco** [comment]
I need help from anybody else, please someone help me!!!

**Reply by mattheuswhale** [reply]
I have a samsung as a backup phone, but I don't know how to check my user agent on that one.

**Reply by AHMNco** [reply]
Dear @fillwithjoy1 can you tell us how we can find a Samsung user agent until @whatthesamuel help us for Samsung user agent

**Comment by AHMNco** [comment]
After months nobody here to help??
I want to have Samsung user agent, popular models like Galaxy S2, S23 or S24
I can't activate VoLTE and Wifi Calling in my Motorola, maybe with Samsung user agent its start to working
Please someone help me
You can try with PC and adb this code to extract Samsung user agent
adb shell dumpsys carrier_config

**Reply by yoro1836** [reply]
I have the S22 Snapdragon version, do you need it?

**Reply by wolerine** [reply]
> I have the S22 Snapdragon version, do you need it?

I need it. I am having similar issue, thus i want to change it.

**Reply by AHMNco** [reply]
@yoro1836 please share it, many people will need it

**Reply by yoro1836** [reply]
Okay

**Reply by yoro1836** [reply]
[Agent.txt](https://github.com/user-attachments/files/16573678/Agent.txt)

**Reply by AHMNco** [reply]
> [Agent.txt](https://github.com/user-attachments/files/16573678/Agent.txt)

Thanks, but I couldn't find user_agent
It's showing in your file
ims.ims_user_agent_string = #MANUFACTURER#_#MODEL#Android#AV##BUILD#
Not mentioned your device info, can you check it and let us help?

**Reply by yoro1836** [reply]
But I want to help, but I don't know exactly how to get it.

**Reply by AHMNco** [reply]
Dear @kyujin-cho please come here as soon as possible and help @yoro1836 to find his Samsung user_agent and share it with everyone
Best Regards

**Comment by AHMNco** [comment]
Dear @kyujin-cho please come here as soon as possible and help us, please

**Comment by fillwithjoy1** [comment]
You know, it's possible that your device may not support VoLTE protocols at all. Just saying
